src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -335,11 +335,17 @@ public virtual void ValidateAsymmetricSecurityKeySize(SecurityKey key, string al
             if (string.IsNullOrEmpty(algorithm))
                 throw LogHelper.LogArgumentNullException(nameof(algorithm));
 
-            int keySize;
-            if (key is JsonWebKey jsonWebKey && jsonWebKey.ConvertedSecurityKey is AsymmetricSecurityKey)
-                keySize = (jsonWebKey.ConvertedSecurityKey as AsymmetricSecurityKey).KeySize;
-            else
-                keySize = key.KeySize;
+            int keySize = key.KeySize;
+            if (key is AsymmetricSecurityKey securityKey)
```

Comment:
I'm not sure why this if-condition is needed.

Reply:
I want to convert from a JsonWebKey before checking the KeySize. Sometimes a JsonWebKey can have 0 as the key size.

Reply:
What I was thinking about is to remove this if-condition and change the line 343 from
`else if (key is JsonWebKey jsonWebKey)` to `if (key is JsonWebKey jsonWebKey)`.
Not sure what is the value of casting a Security Key into AsymmetricSecurityKey and getting its key size if we are doing that already in line 338.

Reply:
@GeoK This is a public method, perhaps we should throw if the key is not Asymmetric.
Perhaps logic of
if key is Asymmetric use keysize
else try to convert to asymmetric
	throw if no conversion